charts/prometheus/templates/ingress.yaml

```
@@ -23,9 +20,7 @@ metadata:
   name: {{ template "prometheus.server.fullname" . }}
   namespace: {{ include "prometheus.namespace" . }}
 spec:
-  {{- if and $ingressSupportsIngressClassName .Values.server.ingress.ingressClassName }}
   ingressClassName: {{ .Values.server.ingress.ingressClassName }}
```

As we are not setting a default value, I'd suggest keeping the conditional:
```suggestion
  {{- if .Values.server.ingress.ingressClassName }}
  ingressClassName: {{ .Values.server.ingress.ingressClassName }}
  {{- end }}
```

I would also uncomment the field `.server.ingress.ingressClassName` in `values.yaml` and set it to an empty string. We should also remove the comments mentioning k8s version 1.18 in `server.ingress`.

---

done